explain-flags.html

```diff
@@ -20,10 +20,15 @@
     function explainFlags() {
         var flagValue = parseInt(document.getElementById('tb').value); //returns 0 or NaN if can't parse
         var summary = "";
+        var pairedRead = lstFlags[0][1] & flagValue;
         for(var i = 0; i < lstFlags.length; i++) {
             var checkbox = document.getElementById('cb' + i)
             if(lstFlags[i][1] & flagValue) {
-                summary  += " &nbsp; &nbsp; " + lstFlags[i][0] + "<br>";
+                if (!pairedRead && i < 8){
```

Comment:
What's special about 8? it looks as though you are saying that if a read is not paired than none of the first 8 flags can be set. the table below shows where this is true:

flag | not valid without 0x1
-----|-----
|0x1 | yes (but impossible)
0x2 | yes
0x4 | no
0x8 | yes
0x10 | no
0x20 | yes
0x40 | yes
0x80 | yes

In the Sam spec this is described by the line 
>  If 0x1 is unset, no assumptions can be made about 0x2, 0x8, 0x20, 0x40 and 0x80.

which Picard took to mean "invalid"

the "OR" of these flags is 234 (decimal) so you can use that in the code (or use binary 0b11101011)

```
...
if (!pairedRead && (lstFlag[i][1] & 235)) {
...
}

Reply:
This is why I shouldn't code when I'm tired...

I looked at the `switchToMate()` function and saw:

https://github.com/broadinstitute/picard/blob/56f200e7fbb332bc9de788f8021495b4d739a241/explain-flags.html#L48

I thought that those must be all the flags assuming 0x1 😅  I can fix this, no problem!